deeppavlov/models/morpho_tagger/common.py

```diff
@@ -125,6 +103,45 @@ def __call__(self, X: List[Union[str, List[str]]],
         Returns:
             a list of prettified morphological analyses
         """
-        return [prettify(x, y, return_string=self.return_string,
-                         begin=self.begin, end=self.end, sep=self.sep)
-                for x, y in zip(X, Y)]
+        return [self.prettify(x, y) for x, y in zip(X, Y)]
+
+    def prettify(self, sent: Union[str, List[str]], tags: List[str]) -> Union[List[str], str]:
+        """Prettifies output of morphological tagger.
+
+        Args:
+            sent: source sentence (either tokenized or not)
+            tags: list of tags, the output of a tagger
+
+        Returns:
+            the prettified output of the tagger.
+
+        Examples:
+            >>> sent = "John really likes pizza"
+            >>> tags = ["PROPN,Number=Sing", "ADV",
+            >>>         "VERB,Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin",
+            >>>         "NOUN,Number=Sing", "PUNCT"]
+            >>> prettifier = TagOutputPrettifier(mode='basic')
+            >>> self.prettify(sent, tags)
+                1	John	PROPN	Number=Sing
+                2	really	ADV	_
+                3	likes	VERB	Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin
+                4	pizza	NOUN	Number=Sing
+                5	.	PUNCT	_
+            >>> prettifier = TagOutputPrettifier(mode='ud')
+            >>> self.prettify(sent, tags)
+                1	John	_	PROPN	_	Number=Sing	_	_	_	_
+                2	really	_	ADV	_	_	_	_	_	_
+                3	likes	_	VERB	_	Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin	_	_	_	_
+                4	pizza	_	NOUN	_	Number=Sing	_	_	_	_
+                5	.	_	PUNCT	_	_	_	_	_	_
+        """
+        if isinstance(sent, str):
+            words = NLTKMosesTokenizer()([sent])[0]
+        else:
+            words = sent
```

Comment:
Tokenization should be made explicit in the pipeline, it's not prettifiers task to tokenize sentences.